src/compile.c

```diff
@@ -357,14 +365,25 @@ static int block_bind_subblock(block binder, block body, int bindflags, int brea
       // a break whenever we come across a STOREV of *anonlabel...
       break_distance++;
     }
+
+    i->any_unbound = (i->symbol && !i->bound_by);
```

Comment:
Do we not need to check the state of the `subfn` and `arglist` sub-trees too here?

Reply:
We do, which is why we pass a pointer to it down into the block_bind_subblock_inner calls on subfn and arglist. It gets set to 1 at L377 in those calls.